astropy/utils/masked/core.py

```diff
@@ -680,73 +780,22 @@ def __ne__(self, other):
         )
         return result.any(axis=-1)
 
-    def _combine_fields(self, mask):
-        masks = []
-        for name in mask.dtype.names:
-            m = mask[name]
-            if m.dtype.names is not None:
-                m = self._combine_fields(m)
-            if m.ndim > mask.ndim:
-                m = m.any(axis=tuple(range(mask.ndim, m.ndim)))
-            masks.append(m)
-        return self._combine_masks(masks, copy=False)
-
-    def _combine_masks(self, masks, out=None, where=True, copy=True):
-        """Combine masks, possibly storing it in some output.
-
-        Parameters
-        ----------
-        masks : tuple of array of bool or None
-            Input masks.  Any that are `None` or `False` are ignored.
-            Should broadcast to each other.  For structured dtype,
-            an element is considered masked if any of the fields is.
-        out : output mask array, optional
-            Possible output array to hold the result.
-        where : array of bool, optional
-            Which elements of the output array to fill.
-        copy : bool optional
-            Whether to ensure a copy is made. Only relevant if a single
-            input mask is not `None`, and ``out`` is not given.
-        """
-        # Simplify masks, by removing empty ones and combining possible fields.
-        masks = [
-            m if m.dtype.names is None else self._combine_fields(m)
-            for m in masks
-            if m is not None and m is not False
-        ]
-        if not masks:
-            if out is None:
-                return False
-            else:
-                # Use copyto to deal with broadcasting with `where`.
-                np.copyto(out, False, where=where)
-                return out
-
-        if len(masks) == 1:
-            if out is None:
-                return masks[0].copy() if copy else masks[0]
-            else:
-                np.copyto(out, masks[0], where=where)
-                return out
-
-        # [...] at the end to ensure we have an array, not a scalar, and
-        # thus can be used for in-place changes in the loop.
-        out = np.logical_or(masks[0], masks[1], out=out, where=where)[...]
-        for mask in masks[2:]:
-            np.logical_or(out, mask, out=out, where=where)
-        return out
+    @staticmethod
+    def _get_data_and_masks(arrays):
```

Comment:
Since it is so easy with Copilot it's worth a docstring here.
```
        """
        Extracts the data and masks from the given arrays.

        Parameters
        ----------
        arrays : iterable
            An iterable of arrays.

        Returns
        -------
        datas, masks
            A tuple of extracted data arrays and mask arrays, respectively.
        """
```

Reply:
OK, thanks. Used slightly adjusted.